src/Address.svelte

```diff
@@ -2,29 +2,36 @@
   import { onMount } from 'svelte';
   import { link } from 'svelte-routing';
   import { ethers } from 'ethers';
-  import { explorerLink, safeLink } from '@app/utils';
-  import Blockies from '@app/Blockies.svelte';
+  import { safeLink } from '@app/utils';
   import Loading from '@app/Loading.svelte';
   import type { Config } from '@app/config';
+  import type { Registration } from '@app/base/registrations/registrar';
+  import Avatar from "@app/Avatar.svelte";
+  import { getRegistration } from '@app/base/registrations/registrar';
   import { identifyAddress, formatAddress, AddressType } from '@app/utils';
 
   export let address: string;
   export let config: Config;
   export let resolve = false;
   export let noBadge = false;
+  export let noAvatar = false;
   export let compact = false;
 
   let checksumAddress = compact
     ? formatAddress(address)
     : ethers.utils.getAddress(address);
+  let loading: boolean = true;
+  let registration: Registration | null = null;
   let addressType: AddressType | null = null;
   let addressName: string | null = null;
 
-  onMount(() => {
-    identifyAddress(address, config).then(typ => addressType = typ);
+  onMount(async () => {
     if (resolve) {
-      config.provider.lookupAddress(address).then(name => addressName = name);
+      addressName = await config.provider.lookupAddress(address);
+      if (addressName) registration = await getRegistration(addressName, config);
     }
+    addressType = await identifyAddress(address, config);
+    loading = false;
```

Comment:
A cleaner pattern is to assign the contents of `onMount` to a promise, eg.

```
let getInfo: Promise<AddressType> = identifyAddress(...);
```

and then in the HTML, you do

```svelte
{#await getInfo}
<!-- Loading -->
{:then info}
<!-- Loaded! -->
{:catch err}
<!-- Err -->
{/await}
```

Reply:
One other thing to consider is that the address type doesn't depend on the name or registration, so it shouldn't have to wait for the others to return, it can be loaded in parallel, for eg. by using the `.then(...)` syntax.

Reply:
> A cleaner pattern than the `loading: boolean` one is to assign the contents of `onMount` to a promise, eg.

Yes I agree I started using it in the users/View page, will apply it also here. 👍

Reply:
3dda56ee20ec6874230e28215b18dca7a3860a05 fixes this